commons-test/src/main/java/org/infinispan/commons/test/PolarionJUnitTest.java

```diff
@@ -60,4 +56,25 @@ long elapsedTime() {
    int numberOfExecutions() {
       return successes.get() + failures.size();
    }
+
+   boolean exceptionsOkay(List<Throwable> throwables) {
+      return throwables.stream().allMatch(t -> {
+         do {
+            if (exceptionOkay(t)) {
+               return true;
+            }
+         } while ((t = t.getCause()) != null);
+         return false;
+      });
+   }
+
+   boolean exceptionOkay(Throwable t) {
+      if (t instanceof IllegalStateException) {
```

Comment:
why not create a `MagicKeyException`? or better, `IgnoreFailureException`?

Reply:
To what end sorry? Do you mean to change the detection so it just looks for a IgnoreFailureException instead of the exception and class name? If so that seems fine to me, but still is the same issue that Ryan originally mentioned.